charactersheet/charactersheet/viewmodels/dm/party.js

```diff
@@ -21,26 +21,20 @@ function PartyViewModel() {
     self.handlePCard = function(inputPCard) {
         var chat = ChatServiceManager.sharedService();
         if (chat.currentPartyNode == null) { return; }
-        var publisherJid;
         var isNewPlayer = true;
-        inputPCard.forEach(function(field, idx, _) {
-            if (field.name === 'publisherJid') {
-                publisherJid = field.value;
+        var newPCard = pCard.fromEntries(inputPCard);
+        var publisherJid = newPCard.get('publisherJid')[0];
+        var pCardInParty = false;
+        var players = chat.getOccupantsInRoom(chat.currentPartyNode);
+        players.forEach(function(player, idx, _) {
```

Comment:
See my above comment on how to simplify this code.

**Note** since you seem to be getting/asking the same question "is this player in the party?" a few times in this code, it might be good to make this a named method. Then it's easier to tell what this code is doing.

Perhaps we can add this to the ChatService Manager?

```js
self.isJidInParty = function(jid) {
  var members = self.getOccupantsInRoom(chat.currentPartyNode);
  return members.some(function(member, idx, _) {
    return member == jid;
  });
};
```